### ISSUE TITLE
relative Xpath builder feature?

### ISSUE THREAD

**Issue by bandophahita** [SelectorsHub/SelectorsHubIssues#31]
### Enhancement feature

The ability to create a rel xpath relative to another tag xpath.

Occasionally I'll run into an element that doesn't have good unique identifiers which devtools and even SelectorsHub will generate an rel xpath that looks like this: `//div[4]//div[1]//div[3]//div[1]//div[1]//button[1]`

While this is functionally ok, these types of xpath have a tendency to break over time. (plus they are ugly and complicated to read)  If the page layout changes and or new divs are added the indexing of the xpath will break.  These are fragile xpath locators.

That said, the surrounding siblings might just have enough uniqueness that could be utilized in building a slightly more robust (re: less fragile) xpath. 

Example:
`//h1[contains(text(), "Payroll")]/../..//button[.="Assign Replacement"]`

**TLDR: It would be nice if SelectorsHub had the ability to mark a specific xpath as the anchor to a relative path to another spot.**
Here is an imagined workflow:

- Click on tag in Elements tab to show all the possible selectors.  
  - (e.g. `//h1[contains(text(), "Payroll")]`)
- Click 'mark position' in SelectorsHub.
- Click on another tag in Elements tab relative to the first tag.
- Witness selector created rel xpath which is relative to the first tag. 
  - (e.g. `//h1[contains(text(), "Payroll")]/../..//button[.="Assign Replacement"]`

It might make sense to show both the regular 'rel xpath' as well as the 'relative to first tag' selector.  This way the user would be able to determine which selector was best for them.

**Comment by selectorshubsanjay** [comment]
Thank you for the feature enhancement. This is great feedback.
I am working on this relative xpath feature with respect to other element. It will be available with SelectorsHub 4.0. Thank you.

**Comment by selectorshubsanjay** [comment]
Now you can generate the Relative XPath with axes in SelectorsHub. This feature is introduced in SelectorsHub 4.0.
Please update and enjoy it. Checkout this tutorial "How to generate Relative Axes based xpath"
https://youtu.be/cU6IKoNmQBE
